src/arch/cortex-m0/syscalls.S

```diff
@@ -13,12 +13,25 @@
 .thumb_func
```

Comment:
All of these changes are now incorrect. The kernel is responsible for saving `r4-r7` (when we preempt we need to save them anyway, and in that case we don't get the application's cooperation)